### PR TITLE
Add doc for task continuous deployment

### LINKED ISSUE #237: Add CD guide for Tasks
As an SCDF user, I'd like to continuously deploy and launch and newer versions of the Tasks from SCDF, so if there's a guide on how to do that, it would be of tremendous help. 

**Acceptance:**
- Similar to [Stream Continuous Delivery](https://dataflow.spring.io/docs/2.5.0.SNAPSHOT/stream-developer-guides/continuous-delivery/), add a new guide under the "Batch Developer guides"
- Perhaps consider moving a portion of the CD docs from the ref. guide to here, and cross-link the microsite link in the ref. guide instead
- Highlight the limitations in CF and K8s, if there's any

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add example for task continuous deployment
 - Copy the existing content from SCDF doc

Resolves #237

### CODE REVIEW

File: content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/0-index.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+summary: true
+---
+
+# Continuous Deployment
+
+Spring Cloud Data Flow provides native support for continuous deployment of task applications.
+
+The application registry in Spring Cloud Data Flow lets you register multiple versions for the same task application.
+
+When launch the task applications, you can choose which version of the task application you want to use for the launch. You can also re-use and update the deployment properties used from the previous launch of the same task.
```
Comment (sabbyanandan) [inline content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/0-index.md:14]:
```suggestion
When launching the task applications, you can choose which version of the task application you want to use for the launch. You can also re-use and update the deployment properties used from the previous launch of the same task.
```

File: content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/1-cd-basics.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+title: 'Continuous Deployment of task applications'
+description: 'Continuous Deployment of Task applications'
+---
+
+# Continuous Deployment of task applications
+
+As task applications evolve, you want to get your updates to production.
+The change can be a new version of the task application that fixes a bug or setting a deployment property different from the previous task launch.
+
+When a task application is registered, a version is associated with it. A task application can have multiple versions associated with it, with one selected as the default. The following image illustrates an application with multiple versions associated with it (see the timestamp entry).
```
Comment (sabbyanandan) [inline content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/1-cd-basics.md:12]:
```suggestion
When a task application is registered in SCDF, a version is typically associated with it. A task application can have multiple versions associated with it, with one selected as the default. The following image illustrates an application with multiple versions associated with it (see the timestamp entry).
```

File: content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/1-cd-basics.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+<img src="images/scdf-task-application-versions.png" alt="Registering Task Applications with multiple Versions" width="740"/>
+
+Versions of an application are managed by registering multiple applications with the same name and coordinates, _except_ the version. For example, if you were to register an application with the following values, you would get one application registered with two versions (2.0.0.RELEASE and 2.1.0.RELEASE):
```
Comment (sabbyanandan) [inline content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/1-cd-basics.md:16]:
```suggestion
Versions of an application are managed in SCDF by registering multiple applications with the same name and coordinates, _except_ the version will be different. For example, if you were to register an application with the following values, you would get one application registered with two versions (2.0.0.RELEASE and 2.1.0.RELEASE):
```

File: content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/1-cd-basics.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  ** Type: `task`
+  \*\* URI: `maven://org.springframework.cloud.task.app:timestamp-task:2.1.0.RELEASE`
+
+Besides having multiple versions, Spring Cloud Data Flow needs to know which version to run on the next launch. This is indicated by setting a version to be the default version. Whatever version of a task application is configured as the default version is the one to be run on the next launch request. You can see which version is the default in the UI as this image shows:
+
+<img src="images/scdf-task-default-version.png" alt="Task Application Default Version" width="740"/>
+
+## Task Launch Lifecycle
+
+In previous versions of Spring Cloud Data Flow, when the request to launch a task was received, Spring Cloud Data Flow would deploy the application (if needed) and run it. If the application was being run on a platform that did not need to have the application deployed every time (CloudFoundry for example), the previously deployed application was used. This flow has changed in 2.3. The following image shows what happens when a task launch request comes in now:
```
Comment (sabbyanandan) [inline content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/1-cd-basics.md:33]:
```suggestion
Before the CD support for Tasks in SCDF, when the request to launch a task was received, Spring Cloud Data Flow would deploy the application (if needed) and run it. If the application was being run on a platform that did not need to have the application deployed every time (CloudFoundry for example), the previously deployed application was used. This flow has changed starting from 2.3. The following image shows what happens when a task launch request comes in now:
```

File: content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/0-index.md (hunk 1 above)
Comment (cppwfs) [reply content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/0-index.md:14]: So sorry about getting to this so late.   `When reading the line  You can also re-use and update the deployment properties used from the previous launch of the same task.`    It should read something like:
`Data Flow re-uses the deployment properties used from the previous launch of the same task.  You can update these properties for each launch.`   If you agree I can updated on merge.

**Comment by cppwfs** [comment]
I think we need to have troubleshooting moved to 4 instead of having it at 3.   i.e. it would be 
1 Getting Started
2 Batch Development
3. Continuous Deployment
4. Troubleshooting

Instead of 
1 Getting Started
2 Batch Development
3. Troubleshooting
4. Continuous Deployment

That way troubleshooting is the last thing you'd need to lookup.    I can change on merge if you agree.

File: content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/1-cd-basics.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
+---
+path: 'batch-developer-guides/continuous-deployment/cd-basics/'
+title: 'Continuous Deployment of task applications'
+description: 'Continuous Deployment of Task applications'
```
Comment (cppwfs) [inline content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/1-cd-basics.md:4]: Update the description to be something different from the title.   i.e. This section will discuss how to use Continuous Deployment of Tasks in SCDF.
If you agree I can update on merge.

Comment (ilayaperumalg) [reply content/documentation/pages/4-batch-developer-guides/4-continuous-deployment/1-cd-basics.md:4]: Yeah, that is better. Please go ahead!

**Pull request closed without merging**